contribute-open-source/pull-request.md

```diff
@@ -1,170 +1,170 @@
+::: 
+ 
+### Step 2: Setup your pull request targets (head and base) 
+ 
+:::{figure} /images/github/github-pr-targets-head-base.png 
+When you setup your pull request, you will need to specify the head and the base. The head refers to the branch that is a-HEAD of the parent repository--ie a branch on your fork! The base is the parent repository that you wish to submit changes to. 
+::: 
+ 
+When creating a **pull request (PR)**, you need to define where your changes should be added and where they come from.  
+ 
+- **<i class="fa-solid fa-database" style="color: #81c0aa;"></i> Base:** The repository where you want your changes to be merged. *(This is usually the original repo you forked.)*   
+- **<i class="fa-solid fa-code-branch" style="color: #81c0aa;"></i> Head:** The repository containing your changes. *(This is your fork. The copy of the repo that you own, where you made edits.)*   
+ 
+ 
+> **🔹 Quick way to remember:** The **head** is "ahead" of the base, meaning it has new changes that the base repository does **not** yet have. 
+ 
+In this example, you want to update: 
+ 
+* **base**: `pyopensci/pyos-demo-package-contribute` with 
+* **head**: commits in your fork `your-username/pyos-demo-package-contribute`. 
+ 
+ 
+> **<i class="fa-solid fa-code-branch" style="color: #81c0aa;"></i> Social Tip:**   
+> Choosing the correct **head** and **base** repositories ensures that your changes  
+> go to the right place. If you accidentally select the wrong base, your PR might not  
+> reach the maintainers you intended. Double-check this before submitting!   
+ 
+ 
+:::{todo} 
+  
+/images/github/base-vs-head.png   
+:alt: "Diagram showing how base and head work in GitHub pull requests."   
+Create Graphic: Simple diagram with arrows from "Your Fork (Head)" → "Original Repo (Base)"   
+ 
+::: 
+ 
+### Step 3: Review your own pull request first 
+ 
+When comparing repositories in a **pull request (PR)**, GitHub provides a **diff view** of changes between files (diff as in difference). Before submitting the pr, carefully review these changes to ensure that what you intended to submit is in the PR (and nothing else). 
+ 
+Since others will review your PR, take time to **clean up unnecessary changes** before submitting. Your time upfront will lead to the PR being merged sooner and will make it easier for others to review.   
+ 
+> **<i class="fa-solid fa-magnifying-glass" style="color: #81c0aa;"></i> Social cue:**   
+> A PR isn't just about submitting changes—it’s about **communicating clearly** with  
+> maintainers. A well-reviewed PR with **only relevant changes** is easier to merge.  
+> **Check that your PR doesn’t include unintended file modifications.**   
+ 
+Before you submit your PR, check: 
+ 
+1. **The number of files:** Do they match what you intended to modify? 
+2. **Review changes in each file:** Are all modifications correct and expected? 
+ 
```

Comment:
Maybe out of scope for a first time, text-only PR that is presumably being done here. But another pre-PR check would be to make sure that the changes are mergable; that is, there are no merge conflicts. The head branch doesn't have to be totally up to date with the base branch, but it helps.

Reply:
another good point. i think it's worth adding that EVEN if that means they have to ask the maintainers why there are conflicts (ie ask for help) .

What are your feeling about the "sync" button where you can sync or hit "update branch" from github . I think it's easiest for beginners. I don't love it myself because it adds a merge commit.

Reply:
I think that advice would be totally appropriate. I also don't use the "update branch" button, but if it works, great!
I will note that I don't think that button will exist if the branch was created without an upstream tracking branch, and even then maybe not if it was created outside of github (in local git). Also, I that button will still not fix merge conflicts, like when the files in the PR were updated since they forked.

Reply:
so many different scenarios. I've updated this lesson. When we eventually teach this we can update the content as needed too!! 🚀 

thank you, as always, for the reviews!!